decidim/decidim
# Fix broken attachments with form validation errors

#### :tophat: What? Why?
When there is a form that has attachments on it (photos or documents), the form will break in case you submit it and the following is true:

- You are updating an existing record
- The existing record has attachments in it
- You are NOT updating the attachments during the record update
- You have updated some other field that causes a validation error on the form

Currently this can happen on the proposals form for instance (see testing).

The following kind of stack trace will appear in the console from this error:

```
ActionView::Template::Error (undefined method `thumbnail_url' for "179":String)
# ./app/views/decidim/proposals/proposals/_edit_form_fields.html.erb:70:in `block in __home_ahukkanen_dev_rails_decidim_decidim_proposals_app_views_decidim_proposals_proposals__edit_form_fields_html_erb__3656523159076971049_129540'
# ./app/views/decidim/proposals/proposals/_edit_form_fields.html.erb:68:in `each'
# ./app/views/decidim/proposals/proposals/_edit_form_fields.html.erb:68:in `__home_ahukkanen_dev_rails_decidim_decidim_proposals_app_views_decidim_proposals_proposals__edit_form_fields_html_erb__3656523159076971049_129540'
# ./app/views/decidim/proposals/proposals/edit.html.erb:20:in `block in __home_ahukkanen_dev_rails_decidim_decidim_proposals_app_views_decidim_proposals_proposals_edit_html_erb___252343347548760021_129320'
# /home/ahukkanen/dev/rails/decidim/decidim-core/app/helpers/decidim/decidim_form_helper.rb:34:in `decidim_form_for'
# ./app/views/decidim/proposals/proposals/edit.html.erb:19:in `__home_ahukkanen_dev_rails_decidim_decidim_proposals_app_views_decidim_proposals_proposals_edit_html_erb___252343347548760021_129320'
# ./app/controllers/decidim/proposals/proposals_controller.rb:186:in `block (2 levels) in update'
# ./app/commands/decidim/proposals/update_proposal.rb:30:in `call'
# ./app/controllers/decidim/proposals/proposals_controller.rb:178:in `update'
# /home/ahukkanen/dev/rails/decidim/decidim-core/app/controllers/concerns/decidim/use_organization_time_zone.rb:21:in `use_organization_time_zone'
# /home/ahukkanen/dev/rails/decidim/decidim-core/app/controllers/concerns/decidim/locale_switcher.rb:24:in `switch_locale'
# ./spec/controllers/decidim/proposals_controller_spec.rb:178:in `block (4 levels) in <module:Proposals>'
# ------------------
# --- Caused by: ---
# NoMethodError:
#   undefined method `thumbnail_url' for "179":String
#   ./app/views/decidim/proposals/proposals/_edit_form_fields.html.erb:70:in `block in __home_ahukkanen_dev_rails_decidim_decidim_proposals_app_views_decidim_proposals_proposals__edit_form_fields_html_erb__3656523159076971049_129540'
```

#### :pushpin: Related Issues
- Related to #7054

#### Testing
- Create a default Decidim development instance with seed data
- Navigate to admin panel to one of the proposals components and enable attachments on it
- Navigate to the front-end to create a new proposal
- Create the proposal WITH ATTACHMENTS until the preview step
- From the preview step, go back to modify the proposal
- Modify the proposal title to a string length of 15 characters exactly (this is another bug that it causes validation error but allows replicating this issue)
- See the error after submission

#### :clipboard: Checklist
:rotating_light: Please review the [guidelines for contributing](../CONTRIBUTING.adoc) to this repository.
